src/stub/parser.rs

```diff
@@ -14,6 +14,7 @@ pub fn parse_generator_stub(generator: &str) -> Result<Stub> {
 /// Exists solely to be consumed with `.parse()`
 struct Parser<'a> {
     token_stream: Box<dyn Iterator<Item = &'a str> + 'a>,
+    pub read_pairings: std::collections::BTreeMap<String, VarType>,
```

Comment:
This probably doesn't need to be `pub` as it's only directly used in a test (and that test would be arguably better if it didn't reach into `read_pairings` directly)

Reply:
```
    let mut parser = Parser::new("join(\"hello\", world)");
    parser.read_pairings.insert(String::from("world"), VarType::Bool);
    let Cmd::WriteJoin { join_terms, .. } = parser.parse_write() else { panic!() };
```
The issue is that the parser would crash with that stub generator (much like it would crash in CG), since it checks if world was initialized the moment it parses the `world` variable.

We can either delete it or rewrite it to parse a `read` command beforehand, like this:

```
    let parser = Parser::new("read world:int\nwrite join(\"hello\", world)");
    let commands = parser.parse().unwrap().commands;
    let Cmd::WriteJoin { join_terms, output_comment: _} = commands[1].clone() else { panic!() };
```

What do you suggest?

Reply:
You can simply remove the `pub` keyword. `parser_tests` is a child module of `parser`, and therefore it has full visibility of everything in `parser`. From the rust book:

> When writing unit tests for a module, it's often a common idiom to have an immediate child of the module to-be-tested named mod test. This module could access any items of the parent module through the second case, meaning that internal implementation details could also be seamlessly tested from the child module.

https://doc.rust-lang.org/reference/visibility-and-privacy.html

Reply:
I'm personally not against reaching into `read_pairings` in the test, but that's a matter preference.

We either
- organically add to `read_pairings` through parsing a read, which will cause this test to break along with the `parse_read` tests, or
- manually add to read pairings, which will cause this test to break along with `parse_variable` if the `read_pairings` implementation is changed

Either way having to remember the state of reads will add an additional dependency to this test no matter what.